src/index.ts

```diff
@@ -125,54 +125,51 @@ export default function createClient<T>(options?: ClientOptions) {
       headers: baseHeaders,
       body: typeof requestBody === 'string' ? requestBody : JSON.stringify(requestBody),
     });
-    const respHeaders = res.headers;
     const response: TruncatedResponse = {
       bodyUsed: res.bodyUsed,
-      headers: respHeaders,
+      headers: res.headers,
       ok: res.ok,
       redirected: res.redirected,
       status: res.status,
       statusText: res.statusText,
       type: res.type,
       url: res.url,
     };
-
-    const contentLength = res.headers.get('Content-Length');
-    const bodyParsed = res.status !== 204 && contentLength !== '0' ? await res.json() : {};
-    return res.ok ? { data: bodyParsed, response } : { error: bodyParsed, response };
+    const body = res.status !== 204 && res.headers.get('Content-Length') !== '0' ? await res.json() : {}; // only parse JSON if status is NOT 204 and Content-Length is NOT '0'
+    return res.ok ? { data: body, response } : { error: body, response };
   }
 
   return {
     /** Call a GET endpoint */
-    async get<U extends PathsWith<T, 'get'>, M extends keyof T[U]>(url: U, init: FetchOptions<T[U][M]>) {
```

Comment:
I still think there’s a better way to do this, but `M extends keyof T[U]` was just wrong. For example, for the `get()` method, the `url` would be filtered correctly and only let you reference paths with a GET operation defined.

However, at the next step, rather than infer the GET types, it would form a union of all that URL’s possible methods. So if it had POST defined, the inference would get confused.

I’ve fixed the issue, but want to make it a little cleaner.